packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import 'framework.dart';
+
+/// Embeds an Android view in the Widget hierarchy.
+///
+/// Embedding Android views is an expensive operation and should be avoided when a Flutter
+/// equivalent is possible.
+///
+/// The Android view object is created using a [PlatformViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewFactory.html).
+/// Plugins can register platform view factories with [PlatformViewRegistry#registerViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html#registerViewFactory-java.lang.String-io.flutter.plugin.platform.PlatformViewFactory-).
+/// Registration is typically done in the plugin's registerWith method, e.g:
+///
+/// ```java
+///   public static void registerWith(Registrar registrar) {
+///     registrar.platformViewRegistry().registerViewFactory("webview", new WebViewFactory(registrar.messenger()));
+///   }
+/// ```
+///
+/// The Android view lifetime is the same as the lifetime of the [State] object for this widget.
+/// When the [State] is disposed the platform view (and auxiliary resources) are lazily
+/// released (some resources are immediately released and some by platform garbage collector).
```

Comment:
This would be a good place to provide an State disposal example or two. You could say it general terms, like: A stateful widget's state is disposed the the widget is removed from the tree or when it is moved within the tree. If the stateful widget has a key and it's only moved relative to its siblings, or it has a [GlobalKey] and it's moved within the tree, it will not be disposed.

Reply:
thanks, copy pasted :smile: